pkg/storage/chunk/prometheus_chunk.go

```diff
@@ -75,23 +95,60 @@ func (p *prometheusXorChunk) Encoding() Encoding {
 	return PrometheusXorChunk
 }
 
-func (p *prometheusXorChunk) Len() int {
+// Wrapper around a Prometheus histogram chunk.
+type prometheusHistogramChunk struct {
+	prometheusChunk
+}
+
+func newPrometheusHistogramChunk() *prometheusHistogramChunk {
+	return &prometheusHistogramChunk{}
+}
+
+func (p *prometheusHistogramChunk) Add(sample model.SamplePair) (EncodedChunk, error) {
+	panic("cannot add sample to histogram chunk")
+}
+
+// AddHistogram adds another histogram to the chunk. While Add works, it is only implemented
+// to make tests work, and should not be used in production. In particular, it appends
```

Comment:
Which tests pass through here? It seems that this code can easily slip into production.

Reply:
None that I can see, added TODO and task to the EPIC: // TODO add unit tests as in chunk_test.go https://github.com/grafana/mimir/issues/3767

Reply:
not sure what you mean, the comment is correct and I've added the todo a bit further up on the whole type
```
// Wrapper around a Prometheus histogram chunk.
// TODO add unit tests as in chunk_test.go https://github.com/grafana/mimir/issues/3767
type prometheusHistogramChunk struct {
	prometheusChunk
}
```